pkg/operator/resource_reconciler.go

```diff
@@ -338,7 +338,7 @@ func (rr *ResourceReconciler) OnUpdate(old, cur interface{}) {
 		return
 	}
 
-	if !k8sutil.IsFinalizerPresent(mCur.GetFinalizers(), k8sutil.StatusCleanupFinalizerName) && rr.DeletionInProgress(mCur) {
+	if !rr.shouldUpdate(mCur) {
```

Comment:
I'm nit-picky here but the shouldUpdate() name doesn't capture the logic of the function which is focused on the deletion state. Maybe we should have a more targeted function in k8sutil returning true if the finalizer is present (e.g. `HasStatusCleanupFinalizer`) and leave the test as-is

```suggestion
	if !k8sutil.HasStatusCleanupFinalizer(mCur) && rr.DeletionInProgress(mCur) {
```